notebooks/02_model/hypertune_spark_deep_dive.ipynb

```diff
@@ -0,0 +1,983 @@
+    "\n",
+    "# Run TrainValidationSplit, and choose the best set of parameters.\n",
+    "model = tvs.fit(train)\n",
+    "\n",
+    "time_spark = time.time() - time_start"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The model parameters in the grid and the best metrics can be then returned. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"estimator: estimator to be cross-validated (current: ALS_4fa58c7f07a708200b53)\\nestimatorParamMaps: estimator param maps (current: [{Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 10, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 1}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 10, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.1}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 10, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.01}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 10, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.001}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 10, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.0001}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 20, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 1}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 20, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.1}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 20, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.01}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 20, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.001}, {Param(parent='ALS_4fa58c7f07a708200b53', name='rank', doc='rank of the factorization'): 20, Param(parent='ALS_4fa58c7f07a708200b53', name='regParam', doc='regularization parameter (>= 0).'): 0.0001}])\\nevaluator: evaluator used to select hyper-parameters that maximize the validator metric (current: RegressionEvaluator_4d2481906fee1984efed)\\nseed: random seed. (default: -6707223660118103754)\""
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model.explainParams()"
```

Comment:
the output directly from this model is pretty ugly, but it can be cleaned up a bit and merged with validation metrics like this:
```
for idx, item in enumerate(model.getEstimatorParamMaps()):
    print('Run {}:'.format(idx))
    print('\tValidation Metric: {}'.format(model.validationMetrics[idx]))
    for key, value in item.items():
        print('\t{0}: {1}'.format(repr(key), value))
```

Reply:
👍